### ISSUE TITLE
Running the Linkwarden v2 migration script crashes with PrismaClientValidationError

### ISSUE THREAD

**Issue by DirgoSalga** [linkwarden/linkwarden#365]
**Describe the bug**
From my docker-compose installation I'm running the script as described in the documentation:

`docker exec -it <MY_CONTAINER_ID> /bin/bash -c 'node scripts/migration/migrateToV2.js'`

The script crashes and return a PrismaClientValidationError, with the following information:

`Unknown arg `pdfPath` in data.pdfPath for type LinkUpdateInput. Did you mean `pdf`? Available args`

**To Reproduce**
1. Start docker-compose
2. Run the script as described in the documentation for docker users.
3. Script crashes.

**Expected behavior**
At this point the script should run the migration.

**Comment by daniel31x13** [comment]
This is because I renamed `pdfPath` and `screenshotPath` column to `pdf` and `image`.

If you replace the current `scripts/migration/migrateToV2.js` script with the code below, it'll most likely work:

```js
const { S3 } = require("@aws-sdk/client-s3");
const { PrismaClient } = require("@prisma/client");
const { existsSync } = require("fs");
const util = require("util");

const prisma = new PrismaClient();

const STORAGE_FOLDER = process.env.STORAGE_FOLDER || "data";

const s3Client =
  process.env.SPACES_ENDPOINT &&
  process.env.SPACES_REGION &&
  process.env.SPACES_KEY &&
  process.env.SPACES_SECRET
    ? new S3({
        forcePathStyle: false,
        endpoint: process.env.SPACES_ENDPOINT,
        region: process.env.SPACES_REGION,
        credentials: {
          accessKeyId: process.env.SPACES_KEY,
          secretAccessKey: process.env.SPACES_SECRET,
        },
      })
    : undefined;

async function checkFileExistence(path) {
  if (s3Client) {
    const bucketParams = {
      Bucket: process.env.SPACES_BUCKET_NAME,
      Key: path,
    };

    try {
      const headObjectAsync = util.promisify(
        s3Client.headObject.bind(s3Client)
      );

      try {
        await headObjectAsync(bucketParams);
        return true;
      } catch (err) {
        return false;
      }
    } catch (err) {
      console.log("Error:", err);

      return false;
    }
  } else {
    try {
      if (existsSync(STORAGE_FOLDER + "/" + path)) {
        return true;
      } else return false;
    } catch (err) {
      console.log(err);
    }
  }
}

// Avatars
async function migrateToV2() {
  const users = await prisma.user.findMany();

  for (let user of users) {
    const path = `uploads/avatar/${user.id}.jpg`;

    const res = await checkFileExistence(path);

    if (res) {
      await prisma.user.update({
        where: { id: user.id },
        data: { image: path },
      });
      console.log(`${user.id}`);
    } else {
      console.log(`${user.id}`);
    }
  }

  const links = await prisma.link.findMany();

  // PDFs
  for (let link of links) {
    const path = `archives/${link.collectionId}/${link.id}.pdf`;

    const res = await checkFileExistence(path);

    if (res) {
      await prisma.link.update({
        where: { id: link.id },
        data: { pdf: path },
      });
      console.log(`${link.id}`);
    } else {
      console.log(`${link.id}`);
    }
  }

  // Screenshots
  for (let link of links) {
    const path = `archives/${link.collectionId}/${link.id}.png`;

    const res = await checkFileExistence(path);

    if (res) {
      await prisma.link.update({
        where: { id: link.id },
        data: { image: path },
      });
      console.log(`${link.id}`);
    } else {
      console.log(`${link.id}`);
    }
  }

  await prisma.$disconnect();
}

migrateToV2().catch((e) => {
  console.error(e);
  process.exit(1);
});
```

**Comment by daniel31x13** [comment]
A more easier approach would probably be to initially upgrade to version 2.0 before proceeding with the upgrade to version 2.4.

**Comment by DirgoSalga** [comment]
And how would I do that? I can use your script. That should not be a problem.

Should I expect something else to break from migrating directly to 2.4?

**Comment by daniel31x13** [comment]
no breaking changes, it's just because I forgot to update the migration script.

**Comment by DirgoSalga** [comment]
![image](https://github.com/linkwarden/linkwarden/assets/29740772/df38c661-49be-4d9a-b804-884a5508435d)
It worked! Thank you.

**Comment by daniel31x13** [comment]
No problem! The script right? 
Just wanna make sure so I could create a new release fixing this bug...

**Comment by DirgoSalga** [comment]
The script was right, no additional changes needed. Thanks!

**Comment by daniel31x13** [comment]
There's also another bug where sometimes the background worker just gets stopped.
Anyway both of them are now fixed in [v2.4.1](https://github.com/linkwarden/linkwarden/releases/tag/v2.4.1).
